### PR TITLE
Added timsort implementation in c,cpp and python

### PR DESCRIPTION
**Pull request by Jaseemck**
Added timsort implementation in c,cpp and python. Issue #773

### CODE REVIEW

File: TimSort/timsort.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
+//GSSoC
+//TimSort
+
+#include<stdio.h>  
+
+int minimum(int a, int b)  
+{  
+        if(a<b)  
+        return a;
```
Comment (sridharjajoo) [inline TimSort/timsort.c]: indentation

File: TimSort/timsort.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        return a;   
+        else   
+        return b;   
+}  
+ 
+//MERGE SORT FUNCTION :   
+void merge(int array[], int left, int mid, int right)  
+{  
+        int LeftLength = mid - left + 1, RightLength = right - mid;  
+        int Left[LeftLength], Right[RightLength];
```
Comment (sridharjajoo) [inline TimSort/timsort.c]: either use hungarian notation or camel case

Comment (Jaseemck) [reply TimSort/timsort.c]: I will use the camelCase.... thankYou!

File: TimSort/timsort.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
+#GSSoC
```
Comment (jainaman224) [inline TimSort/timsort.py]: Please fix file and folder name for all files.

File: TimSort/TimSort.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+// TimSort
+
+
+#include<bits/stdc++.h> 
+using namespace std; 
+
+// MERGE SORT FUNCTION : 
+void merge(int array[], int left, int mid, int right) 
+{  
+	int leftLength = mid - left + 1, rightLength = right - mid;
```
Comment (jainaman224) [inline TimSort/TimSort.cpp]: Use 4 space indentation.

**Pull request merged**